File: hardware/platformio/Assembly/Assembly.py
def xor_from_gates(X, Y):
    not_X = int(not X)
    not_Y = int(not Y)
    
    and1 = X and not_Y
    and2 = not_X and Y
    
    Z = and1 or and2
    return int(Z)

# Test all combinations
print("X Y -> Z (XOR)")
for X in [0, 1]:
    for Y in [0, 1]:
        Z = xor_from_gates(X, Y)
        print(f"{X} {Y} -> {Z}")

File: hardware/Assembly/assembly.c
#include <stdio.h>

// XOR using gates: (X AND NOT Y) OR (NOT X AND Y)
int xor_from_gates(int X, int Y) {
    int not_X = !X;
    int not_Y = !Y;

    int and1 = X && not_Y;
    int and2 = not_X && Y;

    int Z = and1 || and2;
    return Z;
}

int main() {
    int X, Y, Z;

    printf("X Y -> Z (XOR)\n");

    for (X = 0; X <= 1; X++) {
        for (Y = 0; Y <= 1; Y++) {
            Z = xor_from_gates(X, Y);
            printf("%d %d -> %d\n", X, Y, Z);
        }
    }

    return 0;
}

